vm/src/protocol/iter.rs

```diff
@@ -0,0 +1,118 @@
+use crate::IntoPyObject;
+use crate::{
+    builtins::iter::PySequenceIterator, PyObjectRef, PyResult, PyValue, TryFromObject,
+    TypeProtocol, VirtualMachine,
+};
+use std::borrow::Borrow;
+use std::ops::Deref;
+
+/// Iterator Protocol
+// https://docs.python.org/3/c-api/iter.html
+#[derive(Debug, Clone)]
+#[repr(transparent)]
+pub struct PyIter<T = PyObjectRef>(T)
+where
+    T: Borrow<PyObjectRef>;
+
+impl PyIter<PyObjectRef> {
+    pub fn into_object(self) -> PyObjectRef {
+        self.0
+    }
+}
+
+impl<T> PyIter<T>
```

Comment:
I honestly have no idea for Check for now. In CPython, it simply check `tp_iternext`. In RustPython, it checks something more than `tp_iternext` includign `__getitem__`. I want to investigate more about it later whether it happened because we are not correctly implementing iterator or mapping protocol or current `TryFromObject for PyIter` is implemented wrong and covering something more than PyIter itself.